inference-engine/src/vpu/graph_transformer/src/stages/split.cpp

```diff
@@ -153,18 +153,57 @@ Stage StageBuilder::addSplitStage(
         const DataVector& outputs) {
     std::vector<DimValues> offsets;
     offsets.reserve(outputs.size());
-
     DimValues curOffset({{axis, 0}});
-    for (const auto& output : outputs) {
-        offsets.emplace_back(curOffset);
-        curOffset.set(axis, curOffset[axis] + output->desc().dim(axis));
-    }
 
-    auto stage = addSplitStage(model, name, layer, std::move(offsets), input, outputs);
+    const auto haveUnusedOutput = [](const DataVector& outputs) {
+        auto predicate = [](const vpu::Data& out) {
+            return out == nullptr;
+        };
+        return std::any_of(outputs.begin(), outputs.end(), predicate);
+    };
+
+    const auto getOutAxisSizes = [](const ie::CNNLayerPtr& layer, const Dim axis) {
+        const auto outDimsSize = layer->outData[0]->getDims().size();
+        const int idx = dimToIeInd(axis, outDimsSize);
+
+        std::vector<size_t> outAxisSizes;
+        outAxisSizes.reserve(outDimsSize);
+        for (const auto& out : layer->outData) {
+            IE_ASSERT(idx <= out->getDims().size());
+            outAxisSizes.push_back(out->getDims()[idx]);
+        }
+        return outAxisSizes;
+    };
+
+    if (haveUnusedOutput(outputs)) {
+        IE_ASSERT(layer != nullptr);
+        const auto outAxisSizes = getOutAxisSizes(layer, axis);
+        vpu::DataVector usedOutputs;
+        for (int i = 0; i < outputs.size(); ++i) {
+            if (outputs[i] != nullptr) {
+                offsets.emplace_back(curOffset);
+                usedOutputs.push_back(outputs[i]);
+            }
+            curOffset.set(axis, curOffset[axis] + outAxisSizes[i]);
+        }
 
-    stage->attrs().set("axis", axis);
+        auto stage = addSplitStage(model, name, layer, std::move(offsets), input, usedOutputs);
 
-    return stage;
+        stage->attrs().set("axis", axis);
+
+        return stage;
+    } else {    //  This branch handles cases when cnnLayer == nullptr
```

Comment:
I agree that we can combine these branches into one piece of code. Just check and throw at the beginning in case we have `outputs[i]==nullptr && layer==nullptr` with some descriptive message and just leave the code above.

Reply:
Changed.